packages/client/src/http-client/http-client.ts

```diff
@@ -0,0 +1,89 @@
+import { ApiOptions } from '..';
+import { CustomDataType } from '@novu/shared';
+
+export class HttpClient {
+  private backendUrl: string;
+  private apiVersion = 'v1';
+  private headers: Record<string, string> = {
+    'Content-Type': 'application/json',
+  };
+
+  constructor(backendUrl: string, options?: ApiOptions) {
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of.  -->
🗒 **note (non-blocking):** Accept options object to keep it extensible.
In future we can easily add support for timeout, custom headers etc easily without breaking changes.‏

Reply:
Let's also add `backendUrl` inside options and default it to `api.novu.co`.

Reply:
Should we also make this change to the `ApiService`? It will be a breaking change
https://github.com/novuhq/novu/blob/1f49b9b96bf459916588482e3489d86bca8ef1f2/packages/client/src/api/api.service.ts
@SokratisVidros